### ISSUE TITLE
Autocomplete erase its value

### ISSUE THREAD

**Issue by juale86** [react-materialize/react-materialize#923]
### Description

Autocomplete erase he input value if I use this.setState inside the onAutocomplete callback but setState works fine

### Steps to Reproduce
Codes:
...
`handleResponsable = (val) => {
    this.setState({
        responsable:val
    });
}`
render():
`<Row>
        {this.state.aux && <Autocomplete options={{ data: this.state.aux, onAutocomplete: this.handleResponsable }} placeholder="Responsable" />}
</Row>`

**Expected behavior:** I click on an option and the option keep is value in the input value.

**Actual behavior:** The input keep the text I wrote to filter options instead text in option I clicked. If i don't pass an object to setState it works... but I can not solve or undersand if I made a mistake

### Versions

**react-materialize:** ^3.4.1
**materialize-css:** ^1.0.0
**react:** ^16.9.0

**Comment by alextrastero** [comment]
Hi @juale86, could you paste a bit more code?

**Comment by juale86** [comment]
Sorry. I've created a new app with create-react-app and I changed the main component to a stateful component like this (I changed de index.html too): 
`class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {}
  }
  render(){
    return <Autocomplete options={{ data: { "Gus Fring": null, "Saul Goodman": null} /*,onAutocomplete:()=>this.setState({hi:2})*/ }} />
  }
}
export default App;`

In this example the issue is not happening. If you uncomment the onAutocomplete function I click an option and the option is not showed in the Autocomplete component.

**Comment by alextrastero** [comment]
Ok, thanks for the follow up, closing for now :)

**Comment by juale86** [comment]
Closed?. why?

**Comment by alextrastero** [comment]
Oh sorry, I must of not understood what the issue with the actual component is then, we really would need some code to reproduce :)

**Comment by juale86** [comment]
My sorry. I thought I can not send code here. Here I go:
// App.js:
import React from 'react'
import {Autocomplete} from 'react-materialize'
import './App.css'

class App extends React.Component {
  constructor(props){
    super(props)
    this.state = {}
  }

  render(){
    return <Autocomplete
              options={
                { data: { "Gus Fring": null, "Saul Goodman": null}
                        ,onAutocomplete:()=>this.setState({hi:2})
                }
              }
            />
  }
}
    
export default App;


// I added index.html
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/css/materialize.min.css">
// inside head tag

// In body:
<script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/js/materialize.min.js"></script>.

I am not sure is enough. Else how can I send you more code?.
Greetings

**Comment by alextrastero** [comment]
Checking the description on this PR, did you bind the method to this?
`this.handleResponsable = this.handleResponsable.bind(this)`

By the way, the code you pasted works for me.

**Comment by juale86** [comment]
The code I pasted does not use another function, but I tried that. Are you sure what is the problem?. The state changes OK, but the option is not showed in the input if I use setState. The problem is if I comment the line:
`,onAutocomplete:()=>this.setState({hi:2})`
The input changes, but the state does not change correctly. If I use it the state changes but the input doesn't change de value (option elected).
